manifest/src/main/java/org/mule/runtime/manifest/internal/DefaultMuleManifest.java

```diff
@@ -0,0 +1,228 @@
+    return getManifestProperty("Supported-Jdks");
+  }
+
+  @Override
+  public String getRecommendedJdks() {
+    return getManifestProperty("Recommended-Jdks");
+  }
+
+  // synchronize this method as manifest initialized here.
+  @Override
+  public synchronized Manifest getManifest() {
+    if (manifest == null) {
+      manifest = new Manifest();
+
+      InputStream is = null;
+      try {
+        // We want to load the MANIFEST.MF from the mule-core jar. Sine we
+        // don't know the version we're using we have to search for the jar on the classpath
+        URL url = AccessController.doPrivileged(new UrlPrivilegedAction());
+
+        if (url != null) {
+          is = url.openStream();
+        }
+
+        if (is != null) {
+          manifest.read(is);
+        }
+      } catch (IOException e) {
+        logger.warn("Failed to read manifest Info, Manifest information will not display correctly: " + e.getMessage());
+      }
+    }
+    return manifest;
+  }
+
+  private static String getManifestProperty(String name) {
+    return get().getManifest().getMainAttributes().getValue(new Attributes.Name(name));
+  }
+
+  static class UrlPrivilegedAction implements PrivilegedAction<URL> {
+
+    private static final Pattern EMBEDDED_JAR_PATTERN = compile("mule[^-]*-[^-]*-embedded");
+    private static final String MANIFEST_PATH = "META-INF/MANIFEST.MF";
+
+    @Override
+    public URL run() {
+      URL result = null;
+      try {
+        Enumeration<URL> e = MuleManifest.class.getClassLoader().getResources(MANIFEST_PATH);
+        result = getManifestJarURL(e);
+        if (result == null) {
+          // if we haven't found a valid manifest yet, maybe we're running tests
+          result = getManifestTestJarURL();
+        }
+      } catch (IOException e1) {
+        logger.warn("Failure reading manifest: " + e1.getMessage(), e1);
+      }
+      return result;
+    }
+
+    URL getManifestJarURL(Enumeration<URL> e) {
+      SortedMap<String, URL> candidates = new TreeMap<>();
+      while (e.hasMoreElements()) {
+        URL url = e.nextElement();
+        if ((url.toExternalForm().contains("mule-core")
+            && !url.toExternalForm().contains("tests.jar")
+            && !url.toExternalForm().contains("mule-core-components"))
+            || url.toExternalForm().contains("mule-runtime-extension-model")
+            || url.toExternalForm().contains("mule-runtime-ee-extension-model")
+            || (EMBEDDED_JAR_PATTERN.matcher(url.toExternalForm()).find() && url.toExternalForm().endsWith(".jar"))) {
+          candidates.put(url.toExternalForm(), url);
+        }
+      }
+      if (!candidates.isEmpty()) {
+        // if mule-core and mule-core-ee jars are present, then mule-core-ee gets precedence
+        for (String candidateKey : candidates.keySet()) {
+          if (candidateKey.contains("mule-core-ee")
+              || candidateKey.contains("mule-runtime-ee-extension-model")) {
```

Comment:
make this look for mule-manifest instead

Reply:
Done.